control/pure_pursuit/src/pure_pursuit/pure_pursuit_lateral_controller.cpp

```diff
@@ -157,6 +203,83 @@ void PurePursuitLateralController::setResampledTrajectory()
     motion_utils::convertToTrajectoryPointArray(*trajectory_resampled_));
 }
 
+void PurePursuitLateralController::averageFilterTrajectory(
+  autoware_auto_planning_msgs::msg::Trajectory & u)
+{
+  if (static_cast<int>(u.points.size()) <= 2 * param_.path_filter_moving_ave_num) {
+    RCLCPP_ERROR(node_->get_logger(), "Cannot smooth path! Trajectory size is too low!");
+    return;
+  }
+
+  autoware_auto_planning_msgs::msg::Trajectory filtered_trajectory(u);
+
+  for (int64_t i = 0; i < static_cast<int64_t>(u.points.size()); ++i) {
+    TrajectoryPoint tmp{};
+    int64_t num_tmp = param_.path_filter_moving_ave_num;
+    int64_t count = 0;
+    double yaw = 0.0;
+    if (i - num_tmp < 0) {
+      num_tmp = i;
+    }
+    if (i + num_tmp > static_cast<int64_t>(u.points.size()) - 1) {
+      num_tmp = static_cast<int64_t>(u.points.size()) - i - 1;
+    }
```

Comment:
This code could be simplified like this?

```
    int64_t num_tmp = param_.path_filter_moving_ave_num;
    if (i - num_tmp < 0) {
      num_tmp = i;
    }
    if (i + num_tmp > static_cast<int64_t>(u.points.size()) - 1) {
      num_tmp = static_cast<int64_t>(u.points.size()) - i - 1;
    }
```

-> 

```
int64_t num_tmp = std::clamp(param_.path_filter_moving_ave_num), 
                             /*min=*/ i, /*max=*/ static_cast<int64_t>(u.points.size()) - i - 1);

```

Reply:
I think, it shouldn't be like this because minimum and maximum values are conditional.

For example: 
```
for ->
    i -> 40, u.points.size() -> 100, param_.path_filter_moving_ave_num -> 15
```
for this condition num_tmp should be 15. However if we declare like you said, it is gonna be 40.

Reply:
Oh, I misunderstood. Thank you!